File: src/2374.go
package main

import "fmt"

func main() {
	fmt.Println(edgeScore([]int{1, 0, 0, 0, 0, 7, 7, 5}))
}

func edgeScore(edges []int) int {
	nodeWithHighestScore := 0
	nodeToScore := make([]int, len(edges))

	for node, edge := range edges {
		nodeToScore[edge] += node
		if nodeToScore[nodeWithHighestScore] < nodeToScore[edge] {
			nodeWithHighestScore = edge
		} else if nodeToScore[nodeWithHighestScore] == nodeToScore[edge] && edge < nodeWithHighestScore {
			nodeWithHighestScore = edge
		}
	}

	return nodeWithHighestScore
}

File: src/2374.cpp
class Solution {
public:
    int edgeScore(vector<int>& edges) {
        int n = edges.size();
        std::vector score(n, 0);

        for(int i=0; i<n; ++i){
            score[edges[i]] += i;
        }

        long long max_score = -1;
        int highest_score = -1;

        for(int i=0; i<n; ++i){
            if(max_score < score[i]){
                max_score = score[i];
                highest_score = i;
            }
        }
    return highest_score;
    }
};
